saleor/graphql/checkout/mutations/checkout_delivery_method_update.py

```diff
@@ -57,7 +58,8 @@ class Arguments:
     class Meta:
         description = (
             "Updates the delivery method (shipping method or pick up point) "
-            "of the checkout." + ADDED_IN_31
+            "of the checkout. "
+            "Updates the checkout shipping_address if it was missing. " + ADDED_IN_31
```

Comment:
When the Click & Collect delivery is selected, it sets the warehouse address (pickup point) as the shipping address.

In the code below it sets the shipping_address to pickup point address always, not only when it's missing?

Reply:
yes, I forgot to update the description to inform that it always updates the address for click-and-collect delivery for the warehouse address. Do you think it should just check a case where `shipping_address` is empty and only then update this field?

Reply:
Each collection point most likely will have a different address, so we need to update the address whenever a delivery method (collection point) is changed. Could be verified with product owner.

Reply:
I'm not sure if this isn't the feature we must still add? I agree that it makes sense that shipping address should be used for calculation (and collection point = shipping address here). 

On the other hand, some of the other places also use this warehouse logic (eg taxes plugin). Maybe first we can merge this, to unify the behavior (warehouse instead billing), then we can revisit this globally (plugins, apps, flat rates) and apply collection points on top of that?

cc @szczecha you were working on that

Reply:
>Each collection point most likely will have a different address, so we need to update the address whenever a delivery method (collection point) is changed. Could be verified with product owner.

@Air-t @maarcingebala to clarify - current solution will not update address when I switch collection point 1 to collection point 2?

So I can
1. Chose a collection point in state with lower tax
2. Change collection point to state with higher tax (where I live in)
3. Order with lower tax, because address was not recalculated?